### PR TITLE
allow to set source Ip in BasicStructure acceptance tests steps

### PR DESCRIPTION
**Pull request by individual-it**
## Description
steps that do HTTP request should be able to change the own source IP

## Related Issue
https://github.com/owncloud/brute_force_protection/issues/12

## Motivation and Context
brute force protection app needs to send requests from different IP addresses

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/BasicStructure.php [hunk 1]
```diff
@@ -860,13 +889,16 @@ public function userHasLoggedInToAWebStyleSessionUsingTheAPI($user) {
...
+		$options = [];
+		if ($this->sourceIpAddress !== null) {
+			$options ['config'] = [
+				'curl' => [
+					CURLOPT_INTERFACE => $this->sourceIpAddress
+				]
+			];
+		}
+		$options ['cookies'] = $this->cookieJar;
+		$request = $client->createRequest($method, $options);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/BasicStructure.php]:
```
$this->getBaseUrl() . $url
```
disappeared.

File: tests/acceptance/features/bootstrap/BasicStructure.php [hunk 2]
```diff
@@ -886,13 +918,16 @@ public function sendingAToWithRequesttoken($method, $url) {
...
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/BasicStructure.php]:
```
$this->getBaseUrl() . $url
```
disappeared.

**Comment by individual-it** [comment]
@phil-davis fixed the issues

**Review by phil-davis — APPROVED** [review]
Assuming CI passes.

**Comment by individual-it** [comment]
backport in #32454

**Pull request merged**